src/market/market.module.ts

```diff
@@ -494,7 +521,7 @@ export class MarketModuleImpl implements MarketModule {
         });
         return agreement;
       } catch (error) {
-        this.logger.debug("Failed to propose agreement, retrying", { error });
+        this.logger.warn("Failed to propose agreement, retrying", { error });
```

Comment:
this can happen often and is not an error, it's totally ok for a provider to reject our offer if they found someone else in the meantime, that's why this method was created in the first place. No need to spam the user with warnings that are not actionable and not something they should be worried about. `debug` level is correct here